drop-in/src/test/java/com/adyen/checkout/dropin/ui/viewmodel/PaymentMethodsListViewModelTest.kt

```diff
@@ -0,0 +1,212 @@
+
+    @BeforeEach
+    fun setup() {
+        configuration = ConfigurationProvider.getDropInConfiguration()
+        amount = Amount(currency = "EUR", value = 1337)
+        paymentMethods = DataProvider.getPaymentMethodsList()
+        storedPaymentMethods = DataProvider.getStoredPaymentMethods().toMutableList()
+        application = Application()
+        viewModel = PaymentMethodsListViewModel(
+            application = application,
+            paymentMethods = paymentMethods,
+            storedPaymentMethods = storedPaymentMethods,
+            order = null,
+            dropInConfiguration = configuration,
+            amount = amount
+        )
+    }
+
+    @Test
+    fun `test remove stored payment method success`() = runTest {
+        viewModel.paymentMethodsFlow.test {
+            val paymentMethods = storedPaymentMethods
+                .mapToStoredPaymentMethodsModelList(configuration.isRemovingStoredPaymentMethodsEnabled)
+            val storedPaymentMethod = paymentMethods[0]
+
+            viewModel.removePaymentMethodWithId(storedPaymentMethod.id)
+
+            with(expectMostRecentItem()) {
+                assertFalse(contains(storedPaymentMethod))
+            }
+        }
+    }
+
+    @Test
+    fun `test get payment method from payment method model success`() {
+        val paymentMethodModelList = paymentMethods.mapToPaymentMethodModelList()
+
+        val paymentMethod = viewModel.getPaymentMethod(paymentMethodModelList[0])
+
+        assertTrue(paymentMethod == paymentMethods[0])
+    }
+
+    @Nested
+    @DisplayName("test if payment method flow")
+    inner class PaymentMethodsType {
+
+        @Test
+        fun `has stored payment methods`() = runTest {
+            viewModel.paymentMethodsFlow.test {
+                with(expectMostRecentItem()) {
+                    assertTrue(filterIsInstance<StoredPaymentMethodModel>().isNotEmpty())
+                }
+            }
+        }
+
+        @Test
+        fun `has payment methods`() = runTest {
+            viewModel.paymentMethodsFlow.test {
+                with(expectMostRecentItem()) {
+                    assertTrue(filterIsInstance<PaymentMethodModel>().isNotEmpty())
+                }
+            }
+        }
+
+        @Test
+        fun `has payment method header `() = runTest {
+            viewModel.paymentMethodsFlow.test {
+                with(expectMostRecentItem()) {
+                    assertTrue(filterIsInstance<PaymentMethodHeader>().isNotEmpty())
+                }
+            }
+        }
+
+        @Test
+        fun `doesn't have gift card payment method model `() = runTest {
+            viewModel.paymentMethodsFlow.test {
+                with(expectMostRecentItem()) {
+                    assertTrue(filterIsInstance<GiftCardPaymentMethodModel>().isEmpty())
+                }
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("test payment method flow when")
```

Comment:
I don't really see the point in these 2 inner classes specifically, they don't seem like they are different as the tested scenarios are all around the `paymentMethodsFlow` and their titles are also not very descriptive.

I suggest if you can find a better way to group them or just remove the inner classes completely.

Reply:
hmmm yeah I see now inner class `PaymentMethodsType` can be combined in one test case 😅